state-chain/chains/src/sol/transaction_builder.rs

```diff
@@ -876,11 +877,82 @@ pub mod test {
+		let expected_serialized_tx = hex_literal::hex!("0146d109812a2d85d9f150d66af9cf7310019c37b37f20f146ec61f5dec83f04e9a3ff989876c7c3aebc3d9dcf6b025b0d9e46b276859bd937b2d37d466f6d6f058001000609f79d5e026f12edc6443a534b2cdd5072233989b415d7596573e743f3e5b386fb5ec7baaea7200eb2a66ccd361ee73bc87a7e5222ecedcbc946e97afb59ec46167417da8b99d7748127a76b03d61fee69c80dfef73ad2d5503737beedc5a9ed4800000000000000000000000000000000000000000000000000000000000000000306466fe5211732ffecadba72c39be7bc8ce5bbc5f7126b2c439b3a4000000031e9528aae784fecbbd0bee129d9539c57be0e90061af6b6f4a5e274654e5bd472b5d2051d300b10b74314b7e25ace9998ca66eb2c7fbc10ef130dd67028293c8c97258f4e2489f1bb3d1029148e0d830b5a1399daff1084048e7bd8dbe9f859a73bdf31e341218a693b8772c43ecfcecd4cf35fada09a87ea0f860d028168e5c27e9074fac5e8d36cf04f94a0606fdd8ddbb420e99a489c7915ce5699e48900060303090c00040400000004000903809698000000000004000502e093040007060001050e030d010106070f00100a010e0d1136b4eeaf4a557ebc00ca9a3b000000000606080f0001020d0e0b08346cb8a27b9fdeaa230100000014000000ffffffffffffffffffffffffffffffffffffffff040000007c1d0f0700ca9a3b00000000013001afd71da9456a977233960b08eba77d2e3690b8c7259637c8fb8f82cf58a1020f05060a0d09030204").to_vec();
 
 		test_constructed_transaction(transaction, expected_serialized_tx);
 	}
 
+	#[test]
+	fn ccm_native_transfer_length_check() {
+		let transfer_param = TransferAssetParams::<Solana> {
+			asset: SOL,
+			amount: TRANSFER_AMOUNT,
+			to: TRANSFER_TO_ACCOUNT,
+		};
+		let env = api_env();
+		let mut ccm_accounts = ccm_accounts();
+		ccm_accounts.additional_accounts = vec![];
+		println!("ccm_accounts: {:?}", ccm_accounts);
+
+		let transaction = SolanaTransactionBuilder::ccm_transfer_native(
+			transfer_param.amount,
+			transfer_param.to,
+			ccm_parameter().source_chain,
+			None,
+			vec![],
+			ccm_accounts,
+			env.vault_program,
+			env.vault_program_data_account,
+			agg_key(),
+			durable_nonce(),
+			compute_price(),
+			TEST_COMPUTE_LIMIT,
+			vec![chainflip_alt()],
+		)
+		.unwrap();
+
+		let serialized_tx = sign_and_serialize(transaction);
+		assert_eq!(serialized_tx.len(), MAX_TRANSACTION_LENGTH - MAX_CCM_BYTES_SOL);
```

Comment:
It's a bit unclear why this number is used to check the length. 
Is it that the `MAX_CCM_BYTES_SOL` is calculated using the MAX - Len(transfer_tx)? Might be good to leave a comment here to elaborate